### DISCUSSION TITLE
Is it possible to use the current button type in custom theme styling?

### DISCUSSION THREAD

**Discussion by andreaselia** [react-native-elements/react-native-elements#3617] [Q&A] [answered]
Hey all! I'm just having a play around with React Native Elements at the moment, and was wondering if it's possible to apply `buttonStyle` differently based on the button type?

```ts
// What I currently have:
const theme = createTheme({
  components: {
    Button: {
      buttonStyle: {
        backgroundColor: 'red',
      },
    },
  },
})

// What I'd like to achieve something like:
const theme = createTheme({
  components: {
    Button: {
      buttonStyle: ({ type }) => ({
        backgroundColor: type == 'outline' ? 'blue' : 'red',
      }),
    },
  },
})
```

Any suggestions would be greatly appreciated.

**Comment by arpitBhalla** [comment, marked as answer]
https://reactnativeelements.com/docs/customization/typescript#extending-rne-default-components

**Reply by andreaselia** [reply]
Thank you, @arpitBhalla! I genuinely took the time to look through the docs but didn't see that.
